### PR TITLE
Cleanup shared.thrift and move report-specific types to report-server

### LINKED ISSUE #849: cleanup shared.thift
`shared.thift` was used to share types between the storage server and the viewer server.
After merging them this file can be cleaned up the the types used only in the server can be moved to the `report_server.thrift`.

### PR DESCRIPTION
**Pull request by whisperity**
Closes #849.

I've also removed all the types that are unused.

### CODE REVIEW

File: api/v6/report_server.thrift [hunk 1]
```diff
@@ -13,6 +13,70 @@ namespace js codeCheckerDBAccess_v6
...
+  2: i64    startCol,
+  3: i64    endLine,
+  4: i64    endCol,
+  5: i64    fileId
+  6: string filePath
+}
+typedef list<BugPathPos> BugPath
+
+// TODO: This seems to be unused?
+struct SuppressBugData {
```
Comment (csordasmarton) [inline api/v6/report_server.thrift]: We should remove it if it is unused.

File: api/v6/report_server.thrift [hunk 2]
```diff
@@ -13,6 +13,70 @@ namespace js codeCheckerDBAccess_v6
...
+typedef list<SuppressBugData> SuppressBugList
+
+struct ConfigValue {
+  1: string checkerName,
+  2: string attribute,
+  3: string value
+}
+typedef list<ConfigValue> CheckerConfigList
+
+enum Severity {
```
Comment (csordasmarton) [inline api/v6/report_server.thrift]: For my eyes it would be better to put every `enum` definition at the top of this file after `const` declaration.

Comment (csordasmarton) [reply api/v6/report_server.thrift]: Why do we need to specify values (0,10, etc.) for this enum?

File: api/v6/report_server.thrift [hunk 3]
```diff
@@ -219,16 +283,16 @@ service codeCheckerDBAccess {
...
+                                   2: bool     fileContent,
                                    3: Encoding encoding)
                                    throws (1: shared.RequestFailed requestError),
 
   // change review status of a bug.
   // PERMISSION: PRODUCT_ACCESS or PRODUCT_STORE
-  bool changeReviewStatus(1: i64 reportId,
-                          2: shared.ReviewStatus status,
-                          3: string message)
+  bool changeReviewStatus(1: i64          reportId,
```
Comment (csordasmarton) [inline api/v6/report_server.thrift]: Why do you use this indentation style:
```
ParamType     variable,
HugeParamType variable2
```
It's our convention in thrift files for now?
In this case refactor all other function to this form (getSeverityCounts, getReviewStatusCounts etc).

File: api/v6/report_server.thrift [hunk 4]
```diff
@@ -377,7 +441,7 @@ service codeCheckerDBAccess {
   // PERMISSION: PRODUCT_STORE
   bool replaceConfigInfo(
                      1: i64 runId,
-                     2: shared.CheckerConfigList values)
+                     2: CheckerConfigList values)
```
Comment (csordasmarton) [inline api/v6/report_server.thrift]: Indent paramters to function opening bracket.

File: api/v6/report_server.thrift (hunk 2 above)
Comment (whisperity) [reply api/v6/report_server.thrift]: I don't know. I just moved the type to this file.

Comment (whisperity) [reply api/v6/report_server.thrift]: Sure, I'll reorganise this, just first merge the other prs editing this file.

File: api/v6/report_server.thrift (hunk 3 above)
Comment (whisperity) [reply api/v6/report_server.thrift]: I don't know if this is a convention or not. We just started doing this some weeks ago, though it certainly looks better.

File: api/v6/report_server.thrift [hunk 5]
```diff
@@ -104,46 +71,106 @@ enum SortType {
...
-  RESOLVED,
-  UNRESOLVED
+struct RunData {
+  1: i64            runId,        // Unique id of the run.
+  2: string         runDate,      // Date of the run last updated.
+  3: string         name,         // Human-given identifier.
+  4: i64            duration,     // Duration of the run (-1 if not finished).
+  5: i64            resultCount,  // Number of results in the run.
+  6: string         runCmd,       // The used check command.
+  // This maps the detection status to its count
```
Comment (csordasmarton) [inline api/v6/report_server.thrift]: Indentation

**Comment by whisperity** [comment]
Let's make sure at least the Linux tests pass so I didn't accidentally use an ORM type instead of a Thrift type somewhere.

**Comment by whisperity** [comment]
Tests are saying it is good to go. https://travis-ci.org/whisperity/CodeChecker/builds/275491224

**Pull request merged**